tx_tree_test.go

```diff
@@ -803,58 +803,110 @@ func TestTx_Notfound_For_BPTSparseIdxMode(t *testing.T) {
 	})
 }
 
-func TestTx_LazyDelete(t *testing.T) {
+func TestTx_ExpiredDeletion(t *testing.T) {
 	bucket := "bucket"
 
-	t.Run("lazy deletion when Get", func(t *testing.T) {
+	t.Run("expired deletion", func(t *testing.T) {
 		runNutsDBTest(t, nil, func(t *testing.T, db *DB) {
 			txPut(t, db, bucket, GetTestBytes(0), GetTestBytes(0), 1, nil)
+			txPut(t, db, bucket, GetTestBytes(1), GetTestBytes(1), 2, nil)
+			txPut(t, db, bucket, GetTestBytes(2), GetTestBytes(2), 3, nil)
 			txGet(t, db, bucket, GetTestBytes(0), GetTestBytes(0), nil)
+			txGet(t, db, bucket, GetTestBytes(1), GetTestBytes(1), nil)
+			txGet(t, db, bucket, GetTestBytes(2), GetTestBytes(2), nil)
+
+			txDel(t, db, bucket, GetTestBytes(2), nil)
 
 			time.Sleep(1100 * time.Millisecond)
 
-			// this entry will be lazy deleted
-			txGet(t, db, bucket, GetTestBytes(0), nil, ErrNotFoundKey)
+			// this entry will be deleted
+			txGet(t, db, bucket, GetTestBytes(0), nil, ErrKeyNotFound)
+			// this entry still alive
+			txGet(t, db, bucket, GetTestBytes(1), GetTestBytes(1), nil)
+
+			time.Sleep(1 * time.Second)
+
+			// this entry will be deleted
+			txGet(t, db, bucket, GetTestBytes(1), nil, ErrKeyNotFound)
 
 			r, ok := db.BTreeIdx[bucket].Find(GetTestBytes(0))
 			require.Nil(t, r)
 			require.False(t, ok)
+
+			r, ok = db.BTreeIdx[bucket].Find(GetTestBytes(1))
+			require.Nil(t, r)
+			require.False(t, ok)
 		})
 	})
 
-	t.Run("lazy deletion when Delete", func(t *testing.T) {
+	t.Run("update expire time", func(t *testing.T) {
 		runNutsDBTest(t, nil, func(t *testing.T, db *DB) {
 			txPut(t, db, bucket, GetTestBytes(0), GetTestBytes(0), 1, nil)
+			time.Sleep(500 * time.Millisecond)
```

Comment:
Is it better that warp this logic into a function? Like TxExpirePutWithTime()?

Reply:
Maybe that's not very flexible? I may have to do several Puts and then excuting Sleep.

Reply:
> Maybe that's not very flexible? I may have to do several Puts and then excuting Sleep.

like lines 875-882